### PR TITLE
Draft: Grey nodes

### PR DESCRIPTION
**Pull request by russHyde**
One of the tasks in WP5 is to create a treeview that has all-grey nodes.
The code introduced here is one way of doing that.
Rather than modifying the tfpscanner source code, this adds some ggiraph CSS that means that when the user hovers over a node, all other nodes in the tree are turned grey and the hovered node turns red.
Unhovering reverts the node colours to present a heatmap of whichever statistic is being presented.

A module was introduced for the presentation of the treeview data/images in the app.

### CODE REVIEW

File: R/module_treeview.R [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      output$treeview = ggiraph::renderGirafe({
+        tooltip_css = paste0(
+          "background-color:black;",
+          "color:grey;",
+          "padding:14px;",
+          "border-radius:8px;",
+          "font-family:\"Courier New\",monospace;"
+        )
+        girafe_options = list(
+          ggiraph::opts_selection(type = "single"),
```
Comment (jr-nicola) [inline R/module_treeview.R:65]: The idea is to have a 4th treeview type which is initially all grey. Then when a user selects a specific mutation from a drop down menu, the nodes which contain that mutation get recoloured to highlight where the mutation occurs.

Initial idea:
- read in (all grey rds file)
- add layer with new coloured points (based on output from ggbuild)
- make widget in app

If I was to add another ggplot layer (before the widget was created), would the `ggiraph::opts_hover_inv()` override those colours?

*I might be able to make a function that just strips the colour out of the existing treeview rds file.

**Review by jr-nicola — COMMENTED** [review]
Would any new colours added to the ggplot before the widget creation get overridden by the new CSS?

File: R/module_treeview.R (hunk 1 above)
Comment (russHyde) [reply R/module_treeview.R:65]: An alternative solution would be to use:

-  `session$sendCustomMessage(type = '<output_id>_set', message = selected_nodes)` (to select nodes based on user-selections in a drop-down menu)
- the ggiraph options `opts_selection` (to set the theme for selected nodes)
-  and `opts_selection_inv` (to set the theme for non-selected nodes)

Comment (jr-nicola) [reply R/module_treeview.R:65]: The sounds like a better approach than my idea - let's go with that. We don't currently have a table of mutations vs nodes with that mutation. I can manually make a small example for a couple of mutations/nodes to get a working example going, if that would be helpful?

Comment (russHyde) [reply R/module_treeview.R:65]: A small example is as follows:

```r
# in the UI code
  # allows the user to select nodes based on cluster ID
        shiny::column(
          3,
          shiny::selectInput(ns("nodeChoice"), "Choose a node", c("1601", "1859"))
        )
```

```r
# in the server
    # read in the treeview chart
      g <- reactive({
        filename <- get_filename(input$widgetChoice)
        readRDS(filename)
      })

          observeEvent(input$nodeChoice, {
        # 20 == 1601
        # 25 == 1859

        # the 'node' column contains integers that define the IDs for graph-nodes in the htmlwidget
        node_map <- g()$data[c("cluster_id", "node")]
        selection_map <- dplyr::filter(
          node_map,
          .data[["cluster_id"]] %in% input$nodeChoice
        )

        session$sendCustomMessage(
          # "<output_id>_set" defines the collection of selected nodes on the htmlwidget
          type = paste0(ns("treeview"), "_set"),
          #   but the nodes in the widget are ID's using strings, not integers
          message = as.character(selection_map[["node"]])
        )
      })
```

This is just an example of how to use the custom-messages to highlight a node.

Would the mutation lookup table include all mutations, or just those mutations that are mentioned in the heatmap?

**Comment by jr-nicola** [comment]
Can this PR be closed, or do we want to keep it open (and use it to convert treeview to modules later)?

**Comment by russHyde** [comment]
@jr-nicola no problem, I'll pull out the module refactoring.

**Comment by russHyde** [comment]
Please see #56 . This PR can be closed now

**Pull request closed without merging**